app/javascript/components/tooltips/Concept.tsx

```diff
@@ -0,0 +1,244 @@
+  }
+
+  return nextState
+}
+
+function dispatch(
+  id = '',
+  dispatcher: React.Dispatch<any>, // eslint-disable-line @typescript-eslint/no-explicit-any
+  source: ShowActionSource,
+  action?: ShowActionValue
+): void {
+  dispatcher([id, dispatcher, source, action])
+}
+
+export const Concept = ({
+  endpoint,
+  parent,
+  hoverRequestToShow,
+  focusRequestToShow,
+  parentSlug = '',
+}: {
+  endpoint: string
+  parent: HTMLElement | null
+  hoverRequestToShow: boolean
+  focusRequestToShow: boolean
+  parentSlug?: string
+}): JSX.Element => {
+  const tooltipId = `concept-tooltip${
+    parentSlug ? `-${parentSlug}` : parentSlug
+  }`
+  // Retrieve the HTML contents from the endpoint
+  const { isLoading, isError, data: html_content } = useQuery(tooltipId, () =>
+    fetch(endpoint).then((res) => res.text())
+  ) as { isLoading: boolean; isError: boolean; data: string }
+
+  const tooltipRef = useRef(null)
+  const [{ show }, showDispatcher] = useReducer(showReducer, {
+    requestHover: false,
+    requestFocus: false,
+    hover: false,
+    focus: false,
+    show: false,
+  })
+
+  const popper = usePopper(parent, tooltipRef.current, {
+    placement: 'bottom',
+    modifiers: [{ name: 'offset', options: { offset: [0, 8] } }],
+  })
+
+  // For the next two useEffects:
+  // If I ignore the exhaustive deps warning and only have requestToShow, it
+  // works.  If I have the exhaustive deps added (popper) it causes inf loop
+  // due to popper reference changing on every render.
+
+  useEffect(() => {
+    if (hoverRequestToShow && popper?.update) {
+      popper.update()
+      dispatch(tooltipId, showDispatcher, 'request-hover', 'show')
+      return
+    }
+
+    setTimeout(
+      () => dispatch(tooltipId, showDispatcher, 'request-hover', 'hide'),
+      150
+    )
+  }, [hoverRequestToShow]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (focusRequestToShow && popper?.update) {
+      popper.update()
+      dispatch(tooltipId, showDispatcher, 'request-focus', 'show')
+      return
+    }
+
+    dispatch(tooltipId, showDispatcher, 'request-focus', 'hide')
+  }, [focusRequestToShow]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Get styles and classes to apply
+  const styles = { ...popper.styles.popper, zIndex: 10 }
+  const classNames = ['c-tooltip', 'c-concept-tooltip']
+  if (!show) {
+    classNames.push('tw-invisible')
+  }
+  if (show) {
+    classNames.push('tw-pointer-events-auto')
+  }
+
+  // Separate render exists because happy path needs to render
+  // 'dangerouslySetInnerHTML' and this does not.
+  if (isLoading || isError) {
+    return (
+      <div
+        ref={tooltipRef}
+        className={classNames.join(' ')}
+        style={styles}
+        {...popper.attributes.popper}
+        role="tooltip"
+      >
+        {isLoading && <Loading />}
+        {isError && <p>Something went wrong</p>}
+      </div>
+    )
+  }
+
+  return (
+    <div
```

Comment:
Should we use `<dialog>` for these?

Reply:
Apparently, we shouldn't because they don't cause a change in focus.

Reply:
I don't know. I thought dialog has a different meaning with semantic HTML tho?